From 5fd1009b6b0a57a3b40f0d4e45f0849be9016772 Mon Sep 17 00:00:00 2001
From: Rahul Kumar <rahul.kumar1@amd.com>
Date: Thu, 12 Dec 2019 17:14:08 +0530
Subject: [PATCH 4334/4736] drm/ttm: bug fix for sproadic hard hang during
 closing(cttl+c) video playback

Added reservation object as shared resource before adding reservation
object to fence.

Signed-off-by: Rahul Kumar <rahul.kumar1@amd.com>
---
 drivers/gpu/drm/ttm/ttm_execbuf_util.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/gpu/drm/ttm/ttm_execbuf_util.c b/drivers/gpu/drm/ttm/ttm_execbuf_util.c
index e73ae0d22897..41a8dcaa1163 100644
--- a/drivers/gpu/drm/ttm/ttm_execbuf_util.c
+++ b/drivers/gpu/drm/ttm/ttm_execbuf_util.c
@@ -202,7 +202,13 @@ void ttm_eu_fence_buffer_objects(struct ww_acquire_ctx *ticket,
 	list_for_each_entry(entry, list, head) {
 		bo = entry->bo;
 		if (entry->shared)
+		{
+			int r;
+			r = reservation_object_reserve_shared(bo->resv);
+			if (r)
+				return;
 			reservation_object_add_shared_fence(bo->resv, fence);
+		}
 		else
 			reservation_object_add_excl_fence(bo->resv, fence);
 		ttm_bo_add_to_lru(bo);
-- 
2.17.1

